### PR TITLE
feat: make more descriptive errors instead of ProviderError

### PR DESCRIPTION
**Pull request by allnil**
#7359
@Rjected 
I found places where BlockExecutionError::ProviderError was used and tried to add more specific errors
waiting for your feedback

### CODE REVIEW

**Review by Rjected — CHANGES_REQUESTED** [review]
We should see if there is another error variant that already exists in the error enum returned by these methods, which would be appropriate instead of introducing these variants. For example, we already have an error in the fetch before the state root - mapping to a string makes this opaque to higher levels of error handling (even though it's human readable).

**Comment by allnil** [comment]
@Rjected 
Hi!
I've slightly refactored the code and hope it's better now. :)
I added `LatestBlock` error to `BlockExecution` error enum which wraps `ProviderError` and used it in both places where we attempt to retrieve the latest block state during the block execution phase.
Additionally, I discovered how the account cache error is handled in OptimismBuilder  and copied the AccountLoadFailed error from there.

**Comment by allnil** [comment]
and error types grew in size

File: crates/interfaces/src/executor.rs [hunk 1]
```diff
@@ -118,6 +115,9 @@ pub enum BlockExecutionError {
     /// Note: this is not feature gated for convenience.
     #[error("execution unavailable for tests")]
     UnavailableForTest,
+    /// Error when fetch latest block state.
```
Comment (Rjected) [inline crates/interfaces/src/executor.rs:118]:
```suggestion
    /// Error when fetching latest block state.
```

**Review by Rjected — CHANGES_REQUESTED** [review]
nice, just one wording nit

**Review by Rjected — APPROVED** [review]
lgtm, thanks!

**Review by mattsse — APPROVED** [review]

**Pull request merged**